test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/PublishIntegrationTest.cs

```diff
@@ -401,5 +401,19 @@ public PublishIntegrationTest(BuildServerTestFixture buildServer)
             Assert.FileExists(result, PublishOutputPath, "ClassLibrary2.Views.dll");
             Assert.FileExists(result, PublishOutputPath, "ClassLibrary2.Views.pdb");
         }
+
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task Publish_WithNoBuild_FailsWithoutBuild()
```

Comment:
> This doesn't seem like a very good test. 

Why do you think so? It might not look straightforward but I feel this is a clean and simple way to make sure we don't call `Compile` in `NoBuild` scenarios (which is the actual bug). Don't you think verifying whether publish copies the same dll or not is unnecessary?

Reply:
I guess we also want to verify that NoBuild publishes the compiled views. Perhaps another test to cover that?

Reply:
Added.